pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -651,6 +651,14 @@ func (c *ConfigMapUnpacker) ensureJob(cmRef *corev1.ObjectReference, bundlePath
 
 		return
 	}
+	// Cleanup old unpacking job and retry
+	if _, isFailed := getCondition(job, batchv1.JobFailed); isFailed {
+		err = c.client.BatchV1().Jobs(job.GetNamespace()).Delete(context.TODO(), job.GetName(), metav1.DeleteOptions{})
```

Comment:
why delete it manually and not set TTL? (https://kubernetes.io/docs/concepts/workloads/controllers/job/#ttl-mechanism-for-finished-jobs). 

I haven't looked into the entire code, but I'm assuming the controller should re-create a new Job incase it has not been unpacked.

Reply:
iirc, the current implementation [requires completed jobs to persist](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/bundle/bundle_unpacker.go#L533) to indicate an unpacked bundle.